commit 61c3450db96dce96ad2b24b4f0b548e6a46d68e5
Author: Florian Weimer <fweimer@redhat.com>
Date:   Tue Dec 17 18:12:03 2024 +0100

    x86: Avoid integer truncation with large cache sizes (bug 32470)
    
    Some hypervisors report 1 TiB L3 cache size.  This results
    in some variables incorrectly getting zeroed, causing crashes
    in memcpy/memmove because invariants are violated.

Conflicts:
	sysdeps/x86/dl-cacheinfo.h
	  (HAVE_TUNABLES still in use downstream)

diff --git a/sysdeps/x86/dl-cacheinfo.h b/sysdeps/x86/dl-cacheinfo.h
index 276442c2f1f03cd6..a65418892d8dc5e2 100644
--- a/sysdeps/x86/dl-cacheinfo.h
+++ b/sysdeps/x86/dl-cacheinfo.h
@@ -960,12 +960,12 @@ dl_init_cacheinfo (struct cpu_features *cpu_features)
 
 #if HAVE_TUNABLES
   /* NB: The REP MOVSB threshold must be greater than VEC_SIZE * 8.  */
-  unsigned int minimum_rep_movsb_threshold;
+  unsigned long int minimum_rep_movsb_threshold;
 #endif
   /* NB: The default REP MOVSB threshold is 4096 * (VEC_SIZE / 16) for
      VEC_SIZE == 64 or 32.  For VEC_SIZE == 16, the default REP MOVSB
      threshold is 2048 * (VEC_SIZE / 16).  */
-  unsigned int rep_movsb_threshold;
+  unsigned long int rep_movsb_threshold;
   if (CPU_FEATURE_USABLE_P (cpu_features, AVX512F)
       && !CPU_FEATURE_PREFERRED_P (cpu_features, Prefer_No_AVX512))
     {
